grpc4bmi/bmi_client_singularity.py

```diff
@@ -1,25 +1,29 @@
+import logging
 import os
+import subprocess
 import time
 from os.path import abspath
-import subprocess
-import logging
+from tempfile import SpooledTemporaryFile
 from typing import Iterable
 
 import semver
 from typeguard import check_argument_types, qualified_name
 
 from grpc4bmi.bmi_grpc_client import BmiClient
+from grpc4bmi.exceptions import DeadContainerException, SingularityVersionException
 
-REQUIRED_SINGULARITY_VERSION = '>=3.6.0'
+REQUIRED_SINGULARITY_VERSION = '3.6.0'
```

Comment:
I am not sure about this. I think that it should be `>3.6.0`. I know only one way to test this: installing the singularity locally that takes some time and effort. Is there another way? if not, I will make a separate issue and test it.

Reply:
At https://github.com/eWaterCycle/grpc4bmi/blob/6001693b17a303dba46d92219464eb6917b0aeee/grpc4bmi/bmi_client_singularity.py#L24 I do the smaller than comparison. 

To test I did

```pycon
In [1]: import semver

In [2]: semver.VersionInfo.parse('3.5.0') < '3.6.0'
Out[2]: True

In [3]: semver.VersionInfo.parse('3.6.0') < '3.6.0'
Out[3]: False

In [4]: semver.VersionInfo.parse('3.8.0') < '3.6.0'
Out[4]: False
```

Reply:
On cartesius, the singularity version is `3.8.0-1.el7_9`.

Reply:
That also works
```
In [5]: semver.VersionInfo.parse('3.8.0-1.el7_9'.replace('_', '-')) < '3.6.0'
Out[5]: False
```
 Underscores are replaced at [line above if](https://github.com/eWaterCycle/grpc4bmi/blob/6001693b17a303dba46d92219464eb6917b0aeee/grpc4bmi/bmi_client_singularity.py#L23)